sdk/textanalytics/azure-ai-textanalytics/src/test/java/com/azure/ai/textanalytics/TextAnalyticsAsyncClientTest.java

```diff
@@ -0,0 +1,175 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.ai.textanalytics;
+
+import com.azure.ai.textanalytics.models.DetectedLanguage;
+import com.azure.ai.textanalytics.models.Error;
+import com.azure.core.exception.HttpResponseException;
+import com.azure.core.util.Context;
+import org.junit.jupiter.api.Test;
+import reactor.test.StepVerifier;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+public class TextAnalyticsAsyncClientTest extends TextAnalyticsClientTestBase {
+    private TextAnalyticsAsyncClient client;
+
+    @Override
+    protected void beforeTest() {
+        client = clientSetup(httpPipeline -> new TextAnalyticsClientBuilder()
+            .endpoint(getEndPoint())
+            .pipeline(httpPipeline)
+            .buildAsyncClient());
+    }
+
+    /**
+     * Verify that we can get statistics on the collection result when given a batch input with options.
+     */
+    @Test
+    public void detectLanguagesBatchInputShowStatistics() {
+        detectLanguageShowStatisticsRunner((inputs, options) -> {
+            StepVerifier.create(client.detectBatchLanguagesWithResponse(inputs, options))
+                .assertNext(response -> validateBatchResult(response.getValue(), getExpectedBatchDetectedLanguages(), "Language"))
```

Comment:
Since you're doing a switch-case on this arbitrary string "Language" down in validateDocuments, you should make this either a constant string that is shared by all your test cases, or an enum.

Reply:
If we write "language", the switch-case no longer passes.